addons/mail/static/src/components/attachment_viewer/attachment_viewer.js

```diff
@@ -75,6 +77,7 @@ class AttachmentViewer extends Component {
      */
     patched() {
         this._handleImageLoad();
+        this._handlePdfButtons();
```

Comment:
Is your method properly handling being called on a element on which it was already called before?
I have the feeling this will currently generate duplicate stuff?

Because while it's necessary to have it in `patched` for when the `iframe` does appear for the first time/change, it can also happen that `patched` is called for another reason than a change of `iframe`, leading to potentially calling the method multiple times on the same `iframe`.

Reply:
Correct but quite complicated to fix. Maybe we could add a state for this...

Reply:
Fixed with dataset